calculation_scripts/OptimizationWithWrapper.py

```diff
@@ -0,0 +1,92 @@
+import dda_model
+from dda_model import optimizers
+import numpy as np
+import matplotlib.pyplot as plt
+import os
+
+def construct_model():
+    sym_axis = [10.5, 10.5]
+    geometry_shape = [22, 22, 10]
+    pixel_size = 15.0
+    light_direction = [0, 0, 1]
+    light_polarization = [1, 0, 0]
+    wavelength = 542
+    initialization = np.loadtxt("initializations/hourglass.txt")
+    dielectric_constants = [1.01 + 0j, 5.96282 + 3.80423e-7j]
+
+    model = dda_model.DDAModelWrapper(
+        geometry_shape,
+        pixel_size,
+        initialization,
+        dielectric_constants=dielectric_constants,
+        light_direction=light_direction,
+        light_polarization=light_polarization,
+        light_wavelength_nm=wavelength,
+        symmetry_axes=sym_axis,
+        verbose=True,
+    )
+    return model
+
+def saveCurrentStructure(model_, path, iteration):
+    all_parameters = model_.allParameters()
+
+    with open(path + '\\CoreStructure\\CoreStructure' + str(iteration) + '.txt', 'w') as f:
```

Comment:
Same comment as before - apart from the `os.path.join` improvement, this entire loop can be replaced by `np.savetxt`

Reply:
done.